src/components/dashboard/subsettings/AddContact.tsx

```diff
@@ -30,17 +30,23 @@ import { findGateway } from "~gateways/wayfinder";
 import { uploadUserAvatar, getUserAvatar } from "~lib/avatar";
 // import { getAllArNSNames } from "~lib/arns";
 import styled from "styled-components";
-import { useLocation } from "wouter";
 import copy from "copy-to-clipboard";
 import { gql } from "~gateways/api";
 import { useTheme } from "~utils/theme";
+import { useLocation, useSearchParams } from "~wallets/router/router.utils";
+import type { CommonRouteProps } from "~wallets/router/router.types";
 // import { isAddressFormat } from "~utils/format";
 
-interface AddContactProps {
+export interface AddContactDashboardViewProps extends CommonRouteProps {
   isQuickSetting?: boolean;
 }
 
-export default function AddContact({ isQuickSetting }: AddContactProps) {
+export function AddContactDashboardView({
+  isQuickSetting
+}: AddContactDashboardViewProps) {
+  const { navigate } = useLocation();
+  const { address } = useSearchParams<{ address: string }>();
```

Comment:
Another new hook that wraps `wouter`'s `useSearch()` hook. Unfortunately, they only return the unparsed search string.